src/evaluation/Run.jl

```diff
@@ -99,12 +116,18 @@ function run_reactive!(session::ServerSession, notebook::Notebook, old_topology:
 		end
 		
 		cell.running = false
+
+		if !is_resolved(new_topology) && can_help_resolve_cells(new_topology, cell)
+			notebook.topology = new_new_topology = resolve_topology(session, notebook, new_topology, old_workspace_name; skip_reimports=true)
+
+			return run_reactive!(session, notebook, new_topology, new_new_topology, to_run[i+1:end]; deletion_hook=deletion_hook, persist_js_state=persist_js_state, already_in_run=true)
+		end
 	end
-	
+
 	notebook.wants_to_interrupt = false
 	flush_notebook_changes()
 	# allow other `run_reactive!` calls to be executed
-	put!(notebook.executetoken)
+	already_in_run || put!(notebook.executetoken)
```

Comment:
I don't think there need to be a condition here, since there is only one call to reactive_run! which will arrive here because the previous calls return early:
https://github.com/fonsp/Pluto.jl/blob/a208d840d567efe1f431fb5f8dfde98a25b5689e/src/evaluation/Run.jl#L120-L124